### DISCUSSION TITLE
Immich reuploading everything

### DISCUSSION THREAD

**Discussion by Stooovie** [immich-app/immich#14208] [Q&A] [answered]
I had Immich installed and library completely upload a year ago. Lately, probably two or three months ago, Immich decided for some reason to reupload everything. As I use it as a secondary system after iCloud Photos, it takes quite a time, probably in months (56k assets). The source machine is an M4 iPad Pro (on which Immich runs at 6fps tops while uploading). 

Is this something that should happen, maybe because of some sort of refactoring of the code?

Thanks for any tips.

**Comment by alextran1502** [comment]
Did you use this device to upload to Immich previously?

Have you done any kind of system reset on this iPad?

**Comment by Stooovie** [comment]
Hi, I did get a new iPad I use as the source device. Is that what's causing this? 

What's the ideal workflow here? I want to use Immich as a Google Photos replacement alongside iCloud Photos that is (and will remain) my primary platform. There will be source device replacements from time to time. Or maybe use an old iPhone solely for this purpose? (That would be a hack but doable).

**Reply by alextran1502** [reply, marked as answer]
Yes, using a new device will cause this as the device id changed. If you have everything on iCloud already uploaded, you can turn on the option to ignore iCloud asset in the upload settings.

I don't know how did you setup your new device. From my experience of switching iPhone and using the transfer mechanism, the device id is preserved and thus the server knows what assets have been uploaded.

In the future we will use the hash check mechanism so that it doesn't depends on the device id anymore

**Comment by Stooovie** [comment]
Turning on "ignore iCloud" would mean no syncing of future photos. 

Hash checking will probably fix this, thanks for your time!
